src/adjtrans.jl

```diff
@@ -569,3 +569,6 @@ Compute `vec(adjoint(A))`, but avoid an allocating reshape if possible
 """
 _vecadjoint(A::AbstractVector) = vec(adjoint(A))
 _vecadjoint(A::Base.ReshapedArray{<:Any,1,<:AdjointAbsVec}) = adjoint(parent(A))
+
+diagview(A::Transpose, k::Integer = 0) = _vectranspose(diagview(parent(A), -k))
+diagview(A::Adjoint, k::Integer = 0) = _vecadjoint(diagview(parent(A), -k))
```

Comment:
This seems wrong to me.

Shouldn't `diagview` be *guaranteed* to not make a copy?   For example, so that mutating the view mutates the original array?

My understanding is that `_vecadjoint` and `_vectranspose` may make copies.